Document application command index struct + eps, application integration types (#308)
discord-userdoccers/discord-userdoccers: pages/interactions/application-commands.mdx

| description         | string                                                                       | 1-100 character description for `CHAT_INPUT` commands, empty string for `USER` and `MESSAGE` commands | all         |
| options?            | array of [application command option](#application-command-option-structure) | the parameters for the command, max 25                                                                | CHAT_INPUT  |
| default_permission? | boolean (default `true`)                                                     | whether the command is enabled by default when the app is added to a guild                            | all         |

> | integration_types?  | array[[application integration type](#application-integration-types)]        | Contexts where the command can be used, if global; defaults `0`                                       | all         |

Reviewer: ```suggestion
| integration_types?  | array[[application integration type](#application-integration-types)]        | if the command is global, the contexts it can be used in (defaults to `0`)                                       | all         |
```

| USER       | 2    | A UI-based command that shows up when you right click or tap on a user    |
| MESSAGE    | 3    | A UI-based command that shows up when you right click or tap on a message |

###### Application Integration Types

| Name          | Type | Description                                                               |
| ------------- | ---- | ------------------------------------------------------------------------- |

> | GUILD_INSTALL | 0    | Available to be used in guilds                                            |
> | USER_INSTALL  | 1    | Available to be used in a user-app context                                |

Reviewer: ```suggestion
| GUILD_INSTALL | 0    | Available in installed guilds                                            |
| USER_INSTALL  | 1    | Available in Direct Messages and any guild the installed user is in                                |
```

### Endpoints

<RouteHeader method="GET" url="/applications/{application.id}/application-command-index">
  Get Application Command Index
</RouteHeader>

Returns an [Application Command Index](#application-command-index-object) object.

<Alert type="info">

> For authorization, all subsequent endpoints take either a [bot token](/reference#authentication) or [client credentials token](/topics/oauth2#client-credentials-grant) for your application

Reviewer: Are these endpoints in scope for this project?
